x/compliance/handler_test.go

```diff
@@ -1240,6 +1240,74 @@ func TestHandler_CertifyRevokedModel(t *testing.T) {
 	}
 }
 
+func TestHandler_DeleteComplianceInfoForAllCertStatuses(t *testing.T) {
+	setup := Setup(t)
+
+	// add model version
+	vid, pid, softwareVersion, softwareVersionString := setup.AddModelVersion(
+		testconstants.Vid, testconstants.Pid, testconstants.SoftwareVersion, testconstants.SoftwareVersionString)
+
+	// certify model
+	certifyModelMsg := NewMsgCertifyModel(
+		vid, pid, softwareVersion, softwareVersionString, dclcompltypes.MatterCertificationType, setup.CertificationCenter)
+	_, err := setup.Handler(setup.Ctx, certifyModelMsg)
+	require.NoError(t, err)
+
+	deleteComplInfoMsg := NewMsgDeleteComplianceInfo(vid, pid, softwareVersion, dclcompltypes.MatterCertificationType, setup.CertificationCenter)
+	_, err = setup.Handler(setup.Ctx, deleteComplInfoMsg)
+	require.NoError(t, err)
+
+	// check
+	_, err = queryCertifiedModel(setup, vid, pid, softwareVersion, dclcompltypes.MatterCertificationType)
```

Comment:
Please also check (for all three cases: certified, revoked, provision), that
1) queryDeviceSoftwareCompliance works correctly (doesn't return the entry)
2) queryComplianceInfo works correctly (doesn't return the entry)

Reply:
checked